### ISSUE TITLE
Batch Normalization: Why setting updates_collections=None?

### ISSUE THREAD

**Issue by youngleec** [carpedm20/DCGAN-tensorflow#288]
https://github.com/carpedm20/DCGAN-tensorflow/blob/60aa97b6db5d3cc1b62aec8948fd4c78af2059cd/ops.py#L39

In the tensorflow API guides ([tf.contrib.layers.batch_norm](https://www.tensorflow.org/api_docs/python/tf/contrib/layers/batch_norm)), one can set **updates_collections=tf.GraphKeys.UPDATE_OPS** and add the following dependency to the train_op:
```python
update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
with tf.control_dependencies(update_ops):
    train_op = optimizer.minimize(loss)
```
One can also set **updates_collections=None** to force the updates in place. But I didn't find the explicit dependency. Where is that?

Thanks a lot!
